### ISSUE TITLE
Feature request: ability to log custom messages from procmon

### ISSUE THREAD

**Issue by fritzophrenic** [jtpereyda/boofuzz#273]
Possibly related to or superseded by #202 . I would like to add specific logging to my procmon implementation so that I can see more detailed error information than "External instrumentation detects a crash" in the logs.

I can partially workaround this by manually passing the loggers which I crate to my own procmon implementation. However this does not include the database logger, so I am not able to use my log messages for any advanced filtering, etc. through the database.

This could be fixed if the pre_send, post_send, etc. functions received the logger object used by the session.

**Comment by jtpereyda** [comment]
Agreed. Thanks for making this need a little more concrete than the wording in #202. I agree that a situation where the pre_send and post_send functions have a logger object and use the same logging interface as the rest of the code base makes a lot of sense.
